react-with-json-logic/src/components/questionnaireEditor/formEditor/testCaseEditor/ElementEditorTestCase.tsx

```diff
@@ -0,0 +1,185 @@
+        schema={testCaseMetaSchema}
+        formData={getTestCaseMeta(testCase)}
+        onChange={onTestCaseMetaChange}
+        uiSchema={uiSchema}
+      />
+      <Typography>Answers</Typography>
+      <AnswerOrResultEditor
+        availableItems={availableQuestions}
+        currentValues={testCase.answers}
+        onResultItemChange={onAnswerItemChange}
+      />
+      <Typography>Results</Typography>
+      <AnswerOrResultEditor
+        availableItems={availableResults}
+        currentValues={testCase.results}
+        onResultItemChange={onResultItemChange}
+      />
+      <TestCaseResult testResult={runOneTestCase(questionnaireJson, testCase)} />
+    </Grid>
+  );
+}
+
+const AnswerOrResultEditor: React.FC<{
+  availableItems: AvailableItems;
+  currentValues: { [id: string]: any };
+  onResultItemChange: OnItemChange;
+}> = ({ currentValues, availableItems, onResultItemChange }) => {
+  const useStyles = makeStyles(() => ({
+    ul: {
+      listStyleType: "none",
+    },
+  }));
+
+  const classes = useStyles();
+
+  const unusedItemIds = availableItems.map((it) => it.itemId).filter((it) => !Object.keys(currentValues).includes(it));
+  return (
+    <ul className={classes.ul}>
+      {Object.entries(currentValues).map(([id, value]) => (
+        <OneItem
+          currentId={id}
+          currentValue={value}
+          availableIds={[id, ...unusedItemIds]}
+          availableValues={availableItems.find((it) => it.itemId === id)?.possibleValues ?? []}
+          onResultItemChange={onResultItemChange}
+        />
+      ))}
+      <OneItem availableIds={unusedItemIds} availableValues={[]} onResultItemChange={onResultItemChange} />
+    </ul>
+  );
+};
+
+const OneItem: React.FC<{
+  currentId?: string;
+  currentValue?: any;
+  availableIds: string[];
+  availableValues: any[];
+  onResultItemChange: OnItemChange;
+}> = ({ currentId, currentValue, availableIds, availableValues, onResultItemChange }) => {
+  const useStyles = makeStyles((theme) => ({
+    formControl: {
+      margin: theme.spacing(1),
+      minWidth: 240,
+    },
+  }));
+
+  const classes = useStyles();
+
+  return (
+    <li>
+      {currentId !== undefined ? (
+        <FormControl className={classes.formControl}>
+          <InputLabel id={"label-value-" + currentId}>{currentId}</InputLabel>
+          <Select
+            labelId={"label-value-" + currentId}
+            id={"value-" + currentId}
+            value={currentValue}
+            onChange={(event) => onResultItemChange({ itemId: currentId, value: event.target.value as string })}
+          >
+            {availableValues.map((value) => (
+              <MenuItem value={value}>{value}</MenuItem>
+            ))}
+          </Select>
+        </FormControl>
+      ) : null}
+      {currentId === undefined && availableIds.length > 0 ? (
+        <FormControl className={classes.formControl}>
+          <InputLabel id={"label-new-itemId"}>Result Category</InputLabel>
```

Comment:
This needs to be a dynamic title according to content (maybe by property).

At the moment it's also "Result Category" for questions etc.

Reply:
I changed it a bit and added a more generic title. Please check.